docs/specifications/tests/Consistency-TP/consistency05.md

```diff
@@ -1,49 +1,205 @@
-## CONSISTENCY05: Consistency between glue and authoritative data
+# CONSISTENCY05: Consistency between glue and authoritative data
 
-### Test case identifier
+## Test case identifier
 
-**CONSISTENCY05:** Consistency between glue and authoritative data
+**CONSISTENCY05**
 
-### Objective
+## Objective
 
 For name servers that have IP addresses listed as glue, the IP addresses must
-match the authoritative A and AAAA records for that host. This is an IANA
-requirement (https://www.iana.org/help/nameserver-requirements).
+match the authoritative A and AAAA records for that host. This is an IANA 
+[name server requirement].
+
+The objective of this test is to verify that the glue address records 
+in the delegation are consistent with authoritative data.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+1. Obtain the set of name server names from the NS records in the 
+   delegation of *Child Zone* using [Method2] and any glue IP addresses
+   from the same delegation using [Method4].
+
+   1. Extract the [in-bailiwick] name server names and create the set
+      "Delegation Strict Glue", where each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set may be empty.
+
+   2. Extract the [out-of-bailiwick] name server names and create the 
+      set "Delegation Extended Glue", where each name server name 
+      is matched with its IP address or addresses, if available. The 
+      set may be empty.
+
+2. Obtain the set of name server names for the *Child Zone* using
+   [Method2] and [Method3] and extract the [in-bailiwick] name 
+   server names, "IB NS Name Set".
+
+3. Create an empty set of name server name with associated IP address
+   or addresses, "Address Records From Child".
+
+4. Obtain the set of name server IP addresses for *Child Zone* 
+   using [Method4] and [Method5], "NS IP", if *IB NS Name Set* is 
+   non-empty.
+
+5. If *IB NS Name Set* is non-empty, then for each name server name in
+   that set do:
+
+   1. Create one A query and one AAAA query with the RD flag unset
+      and name server name as owner name.
+
+   2. For each name server in *NS IP* and for each record 
+      types (A, AAAA):
+      1. Send the address query to the name server.
+      2. If there is no DNS response from the server, then
+         output *[NO_RESPONSE]*.
+      3. Or, if the response is a delegation (referral) to a 
+         sub-zone of *Child Zone*, then:
+         1. Copy the adress query (A, AAAA) that gave the referral
+            response.
+         2. Set the RD flag in the copied query (from unset to set).
+         3. Do a DNS lookup of the the query using a resolving name server 
+            on the public DNS.
+            * The lookup must take into account changes that
+              undelegated data has created, if any.
+         4. If the lookup returns the relevant address record or records
```

Comment:
This could be clearer. First of all, which one of the following is correct?
* "If the lookup returns the relevant address record or (records with the same owner name as in the query), then"
* "If the lookup returns (the relevant address record or records) with the same owner name as in the query, then"

Second, is "relevant" purely educational here, or does it constitute additional checks? And if so, what additional checks?